etl/steps/data/garden/owid/2022-12-19/gdp_last_2000_years.py

```diff
@@ -0,0 +1,139 @@
+        df[f"{var}_scalar"] = value_ref_countries / df[f"{var}_maddworld"]
+
+        # The reference value is the 1990 WDI value
+        value_ref = df.loc[df["year"] == year_ref, f"{var}_wdi"].iloc[0]
+        # The rest of the estimated values are the division between the reference value and the scalars
+        df[f"{var}_estimated"] = value_ref / df[f"{var}_scalar"]
+
+        # Rename the estimated to gdp/gdp_per_capita
+        # Round depending on the variable
+        if var == "gdp":
+            df[f"{var}"] = np.where(df[f"{var}_wdi"].isnull(), df[f"{var}_estimated"].round(0), df[f"{var}_wdi"])
+        else:
+            df[f"{var}"] = np.where(df[f"{var}_wdi"].isnull(), df[f"{var}_estimated"], df[f"{var}_wdi"])
+
+    # Specify "World" entity for each row
+    df["country"] = "World"
+
+    # Select only country, year and the new gdp and gdp_per_capita columns, filtering the year 1990
+    df = df[["country", "year"] + var_list + ["gdp_maddworld", "gdp_per_capita_maddworld"]]
+
+    # Concatenate WDI results with estimation
+    df = pd.concat([df, df_wdi[df_wdi["year"] > year_ref]], ignore_index=True)
+
+    # Specify new reference year for the 1 - 1820 period
+    year_ref = 1820
+
+    # Keep the data until 1820 in the old Maddison Dataset 2010
+    df_maddison2010 = df_maddison2010[df_maddison2010["year"] <= year_ref].reset_index(drop=True)
+    df_maddison2010 = df_maddison2010.drop(columns=["population"])
+
+    # Merge datasets to include Maddison Dataset 2010
+    df = pd.merge(df, df_maddison2010, on="year", how="outer", suffixes=(None, "_madd2010"), sort=True)
+
+    for var in var_list:
+        # The reference value for Maddison 2010 is the 1820 Maddison 2010 value
+        value_ref_countries = df.loc[df["year"] == year_ref, f"{var}_madd2010"].iloc[0]
+        # The scalar is this reference value divided by each each country aggregation
+        df[f"{var}_scalar"] = value_ref_countries / df[f"{var}_madd2010"]
+
+        # The reference value is the estimated 1820 value
+        value_ref = df.loc[df["year"] == year_ref, f"{var}"].iloc[0]
+        # The rest of the estimated values are the division between the reference value and the scalars
+        df[f"{var}_estimated"] = value_ref / df[f"{var}_scalar"]
+
+        # Rename the estimated to gdp/gdp_per_capita
+        # Round depending on the variable
+        if var == "gdp":
+            df[f"{var}"] = np.where(df[f"{var}"].isnull(), df[f"{var}_estimated"].round(0), df[f"{var}"])
+        else:
+            df[f"{var}"] = np.where(df[f"{var}"].isnull(), df[f"{var}_estimated"], df[f"{var}"])
+    df["country"] = "World"
+
+    df = df[["country", "year"] + var_list]
+
+    # create new dataset with the same metadata as meadow
+    ds_garden = Dataset.create_empty(dest_dir)
+
+    tb_garden = Table(df)
+
+    # update metadata from yaml file
+    ds_garden.metadata.update_from_yaml(N.metadata_path)
```

Comment:
I think you can use the new method `ds_garden.update_metadata`, which will update the dataset metadata, including the metadata of the tables included.

Reply:
I think I had an `Exception: table has no short_name` error and I preferred to switch to the old method. At least that's what I see now when I change it